### PR TITLE
Refs [TASK] [34653] [Handle event setting screen]

### PR DESCRIPTION
**Pull request by huydd-2678**
## Related Tickets
- [#Task 34653: Handle event setting screen](https://edu-redmine.sun-asterisk.vn/issues/34653)
## WHAT
- Turn on/off notification
+create notification
+notify notification every morning
+notification click: open app
- Change language of app:VietNam, English
## Evidence (Screenshot or Video)
https://user-images.githubusercontent.com/77426474/112925124-89724f00-913b-11eb-922b-3a083575026a.mp4
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/li/weatherapp/data/source/local/LanguageLocalDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+) : LanguageDataSource {
+
+    override fun getLanguage(): String = preferencesHelper.get(LANGUAGE, String())
+
+    override fun putLanguage(languageCode: String) {
+        preferencesHelper.put(LANGUAGE, languageCode)
+    }
+
+    companion object {
+        const val LANGUAGE = "language"
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/data/source/local/LanguageLocalDataSource.kt]: sao lại để public thế, mà sao k lưu vào trong constant SharedPrefsKey

File: app/src/main/java/com/li/weatherapp/service/NotificationService.kt [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            setContentIntent(pendingIntent)
+            setCustomBigContentView(remoteViews)
+            setAutoCancel(true)
+            priority = NotificationCompat.PRIORITY_HIGH
+        }
+        notificationManager.notify(NOTIFICATION_ID, builder.build())
+    }
+
+    companion object {
+        private const val CHANNEL_ID = "WeatherNotificationChannel"
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/service/NotificationService.kt]: A nghĩ là nên đặt tên theo job mà nó thực hiện thì dễ quản lý hơn

File: app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt [hunk 3]
```diff
@@ -1,35 +1,102 @@
...
+import com.li.weatherapp.utils.showToast
 import kotlinx.android.synthetic.main.fragment_setting.*
+import java.util.*
 
-class SettingFragment : BaseFragment() {
+@Suppress("DEPRECATION")
+class SettingFragment : BaseFragment(), SettingContract.View {
+
+    private var settingPresenter: SettingContract.Presenter? = null
+    private var currentCityPresenter: CurrentCityContract.Presenter? = null
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt]: Tại sao lại cần 2 presenter thế ?

File: app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt [hunk 4]
```diff
@@ -43,16 +110,40 @@ class SettingFragment : BaseFragment() {
...
+    private fun changeItemLanguageSpinner() {
+        spinnerLanguage.onItemSelectedListener = object :
+            AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>,
+                view: View, position: Int, id: Long
+            ) {
+                val language = settingPresenter?.getLanguage()
+                when (position) {
+                    0 -> {
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt]: hard code

File: app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt (hunk 3 above)
Comment (huydd-2678) [reply app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt]: 1 cái setting contract là phần xử lý language, còn cái currentCity là Liên có tạo lúc làm fragment đầu tiên rồi, e dùng để lấy tọa độ để request API

File: app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt [hunk 5]
```diff
@@ -43,16 +106,40 @@ class SettingFragment : BaseFragment() {
...
+                    VIETNAMESE_SPINNER_POSITION -> {
+                        presenter?.putLanguage(LOCALE_VIETNAM)
+                    }
                 }
+                if (language != presenter?.getLanguage()) {
+                    startActivity(
+                        context?.let {
+                            MainActivity.getIntent(it)
+                                .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
+                        }
```
Comment (huytq-2422) [inline app/src/main/java/com/li/weatherapp/ui/setting/SettingFragment.kt]: nếu context null thì crash app sao ?

**Pull request merged**